org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/TypeHierarchyCommand.java

```diff
@@ -176,7 +177,13 @@ private void resolve(TypeHierarchyItem item, IType type, TypeHierarchyDirection
 		if (monitor.isCanceled() || resolve <= 0) {
 			return;
 		}
-		ITypeHierarchy typeHierarchy = (direction == TypeHierarchyDirection.Parents) ? type.newSupertypeHierarchy(DefaultWorkingCopyOwner.PRIMARY, monitor) : type.newTypeHierarchy(type.getJavaProject(), DefaultWorkingCopyOwner.PRIMARY, monitor);
+		ITypeHierarchy typeHierarchy;
+		if (direction == TypeHierarchyDirection.Parents) {
+			typeHierarchy = type.newSupertypeHierarchy(DefaultWorkingCopyOwner.PRIMARY, monitor);
+		} else {
+			ICompilationUnit[] workingCopies = JavaModelManager.getJavaModelManager().getWorkingCopies(DefaultWorkingCopyOwner.PRIMARY, true);
+			typeHierarchy = type.newTypeHierarchy(workingCopies, monitor);
```

Comment:
Riiiight. Because the two source files are in separate projects. Is this why we were seeing the issue @snjeza  ? `type.getJavaProject()` completely left the other project out.

Reply:
You are right.

Reply:
More specifically, it looks like the old `newTypeHierarchy(IJavaProject, WorkingCopyOwner, IProgressMonitor)` uses just the `IJavaProject` for the search scope while the new `newTypeHierarchy(ICompilationUnit[], IProgressMonitor)` uses a `SearchEngine.createWorkspaceScope()` for the scope. That's the difference.